images/worker/slurmd_entrypoint.sh

```diff
@@ -47,15 +42,26 @@ echo "Complement jail rootfs"
 echo "Waiting until munge is started"
 while [ ! -S "/run/munge/munge.socket.2" ]; do sleep 2; done
 
-echo "Detect available GPUs"
-# The following command converts the nvidia-smi output into the Gres GPU string expected by Slurm.
-# For example, if "nvidia-smi --list-gpus" shows this:
-#   GPU 0: NVIDIA A100-SXM4-80GB (UUID: <...>)
-#   GPU 1: NVIDIA A100-SXM4-80GB (UUID: <...>)
-#   GPU 2: NVIDIA V100-SXM4-16GB (UUID: <...>)
-# the GRES variable will be equal to "gpu:nvidia_a100-sxm4-80gb:2,gpu:nvidia_v100-sxm2-16gb:1".
-# See Slurm docs: https://slurm.schedmd.com/gres.html#AutoDetect
-GRES=$(nvidia-smi --query-gpu=name --format=csv,noheader | sed -e 's/ /_/g' -e 's/.*/\L&/' | sort | uniq -c | awk '{print "gpu:" $2 ":" $1}' | paste -sd "," -)
+GRES=""
+if [ "$SLURM_CLUSTER_TYPE" = "gpu" ]; then
+    echo "Slurm cluster type is - $SLURM_CLUSTER_TYPE Detect available GPUs"
+    # The following command converts the nvidia-smi output into the Gres GPU string expected by Slurm.
+    # For example, if "nvidia-smi --list-gpus" shows this:
+    #   GPU 0: NVIDIA A100-SXM4-80GB (UUID: <...>)
+    #   GPU 1: NVIDIA A100-SXM4-80GB (UUID: <...>)
+    #   GPU 2: NVIDIA V100-SXM4-16GB (UUID: <...>)
+    # the GRES variable will be equal to "gpu:nvidia_a100-sxm4-80gb:2,gpu:nvidia_v100-sxm2-16gb:1".
+    # See Slurm docs: https://slurm.schedmd.com/gres.html#AutoDetect
+    GRES="$(nvidia-smi --query-gpu=name --format=csv,noheader | sed -e 's/ /_/g' -e 's/.*/\L&/' | sort | uniq -c | awk '{print "gpu:" $2 ":" $1}' | paste -sd ',' -)"
+    echo "Detected GRES is $GRES"
+
+    echo "Create NVML symlink with the name expected by Slurm"
+    pushd /usr/lib/x86_64-linux-gnu
+        ln -s libnvidia-ml.so.1 libnvidia-ml.so
+    popd
+else
+    echo "Detect available CPUs"
```

Comment:
The message looks wrong here

Reply:
Ah, OK, just read it as "GPUs", not "CPUs".
Anyway, it's not only about CPUs.

What about printing "Skip GPU detection"?